src/aequitas/flow/methods/preprocessing/label_flipping.py

```diff
@@ -0,0 +1,138 @@
+        self.base_estimator = base_estimator(**args)
+        self.n_estimators = n_estimators
+
+        self.fair_ordering = fair_ordering
+        if ordering_method not in METHODS:
+            raise ValueError(f"Invalid margin method. Try one of {METHODS}.")
+        self.ordering_method = ordering_method
+        self.used_in_inference = False
+        self.seed = seed
+
+    def fit(self, X: pd.DataFrame, y: pd.Series, s: Optional[pd.Series]) -> None:
+        pass
+
+    def _score_instances(self, X, y, estimators):
+
+        if self.ordering_method == "ensemble_margin":
+            scores = pd.Series(dtype=float)
+            y_pred = np.array([clf.predict(X.values) for clf in estimators])
+            for i in X.index:
+                v_1 = y_pred[:,i].sum()
+                v_0 = y_pred.shape[0] - v_1
+                if y.loc[i] == 1:
+                    scores.loc[i] = (v_1 - v_0) / self.n_estimators
+                else:    
+                    scores.loc[i] = (v_0 - v_1) / self.n_estimators
+        
+        elif self.ordering_method == "residuals":
+            y_pred = np.array([abs(y - clf.predict_proba(X.values)[:,1]) for clf in estimators])
+            scores = pd.Series(y_pred.sum(axis=0) / self.n_estimators, index=X.index)
+
+        return scores
+    
+    def _calculate_prevalence_disparity(self, y: pd.Series, s: pd.Series):
+        prevalence_0 = y.loc[s == 0].value_counts()[1] / y.shape[0]
+        prevalence_1 = y.loc[s == 1].value_counts()[1] / y.shape[0]
+
+        return prevalence_0 - prevalence_1
+    
+    def _label_flipping(self, y: pd.Series, s: Optional[pd.Series], scores: pd.Series):
+        y_flipped = y.sort_values(key = lambda x: scores.loc[x], ascending = (self.ordering_method == "ensemble_margin"))
+        n_flip = self.flip_rate*len(y)
+
+        if self.fair_ordering:
+            disparity = self._calculate_prevalence_disparity(y_flipped, s)
+
+            for i in y_flipped.index:
+                if (disparity > 0 and s.loc[i] != y_flipped.loc[i]) or (disparity < 0 and s.loc[i] == y_flipped.loc[i]):
+                    y_flipped.loc[i] = 1 - y_flipped.loc[i]
+                    disparity = self._calculate_prevalence_disparity(y_flipped, s)
+                    n_flip -= 1
+
+                if n_flip == 0:
+                    break
+
+        else:
+            y_flipped[:n_flip] = 1 - y_flipped[:n_flip]
+
+        return y_flipped.reindex(y.index)
+
+    def transform(self, X: pd.DataFrame, y: pd.Series, s: Optional[pd.Series]) -> Tuple[pd.DataFrame, pd.Series, Optional[pd.Series]]:
+        self.logger.info("Transforming data with LabelFlipping.")
+
+        if s is None and self.fair_ordering:
+            raise ValueError("Sensitive Attribute `s` not passed. Must be passed if `fair_ordering` is True.")
+        
+        X_num = pd.get_dummies(X)
+
+        bagging = BaggingClassifier(estimator=self.base_estimator, 
```

Comment:
Since we are fitting stuff here, I think we can do it in the `fit` method too. You have to save the `BaggingClassifier` object as an attribute of this class, but it will be fine, that way you can transform multiple times!